clang/docs/LanguageExtensions.rst

```diff
@@ -2311,6 +2311,30 @@ argument.
   int *pb =__builtin_preserve_access_index(&v->c[3].b);
   __builtin_preserve_access_index(v->j);
 
+``__builtin_unique_stable_name``
+--------------------------------
+
+``__builtin_unique_stable_name()`` is a builtin that takes a type or unevaluated
+expression and produces a string literal containing a unique name for the type
```

Comment:
1. I tried to add this but there is no warning emitted at all ! 
2. Tried to assert that (i==0) and (j==2) but compiler says that assert identifier is not found - trying to work around this

Reply:
The first bullet could be due to a bug, but we'd have to see the test case to know for sure.

You shouldn't use an `assert` directly in the test because the test will change behavior on debug vs release builds. I'd recommend using a CodeGen test for those to see if the increment instructions are emitted or not.

Reply:
I think the side-effect in an unevaluted context warning is incredibly specifically written.  So I didn't implement the warning...

Reply:
That's fair -- @schittir can implement the test case for it and we can add comments around the test case to explain the current behavior vs what we think might be "more standard" behavior, and then address whether it's a bug (or one worth addressing now) in the community review.

Reply:
Added the cases. Please let me know what you think.